### PR TITLE
[1496] URN input box added for lead schools in contact section

### PR DESCRIPTION
**Pull request by stellenberger**
### Context

- [This trello ticket](https://trello.com/c/0qY7Qmqn/1496-s-add-the-ui-for-providers-which-are-lead-school-only-in-order-to-capture-urn)
- Add the input UI for URN to the provider contact page, for schools that are lead_schools only

### Changes proposed in this pull request

- Added input for URN to the provider contact page with supporting tests

### Guidance to review

- Fire up your TTAPI and Publish locally
- Log in as Mary
- Navigate to a provider that you know is a lead school
- Click on Contact to edit the contact details
- You should see an input box for a URN for the said school. If you don't, you may need to check that the school is a type `lead_school`

### Checklist

- [X] Make sure all information from the Trello card is in here
- [X] Attach to Trello card
- [x] Rebased master
- [X] Cleaned commit history
- [X] Tested by running locally
- [ ] Product Review

### CODE REVIEW

**Comment by defong** [comment]
[[S] Add the UI for providers which are `lead school` only, in order to capture URN](https://trello.com/c/0qY7Qmqn/1496-s-add-the-ui-for-providers-which-are-lead-school-only-in-order-to-capture-urn)

File: spec/factories/providers.rb [hunk 1]
```diff
@@ -64,6 +65,10 @@
       accredited_body? { true }
     end
 
+    trait :lead_school do
+      urn { Faker::Number.number(digits: 5) }
```
Comment (defong) [inline spec/factories/providers.rb:69]:
```suggestion
      urn { Faker::Number.number(digits: [5, 6].sample) }
```

File: app/views/providers/contact.html.erb [hunk 2]
```diff
@@ -56,6 +56,13 @@
         <%= f.text_field field, options.merge(class: 'govuk-input govuk-!-width-two-thirds') %>
       <% end %>
 
+      <% if @provider.provider_type == "lead_school" %>
+        <%= render "shared/form_field",
+          form: f, field: :urn, label: "URN (Unique Reference Number)" do |field, options| %>
+          <%= f.text_field field, options %>
```
Comment (avinhurry) [inline app/views/providers/contact.html.erb:62]: Can we have the same size text box as UKPRN, which will be above it please?

<%= f.text_field field, options.merge(class: 'govuk-input govuk-input--width-10') %>

Comment (stellenberger) [reply app/views/providers/contact.html.erb:62]: all done 👍

File: spec/factories/providers.rb (hunk 1 above)
Comment (stellenberger) [reply spec/factories/providers.rb:69]: all done 👍

**Review by defong — CHANGES_REQUESTED** [review]
Umm... you forgot to push?

**Review by defong — APPROVED** [review]

File: app/views/providers/contact.html.erb [hunk 3]
```diff
@@ -55,6 +55,13 @@
         <%= f.text_field field, options.merge(class: 'govuk-input govuk-!-width-two-thirds') %>
       <% end %>
 
+      <% if @provider.provider_type == "lead_school" %>
+        <%= render "shared/form_field",
+          form: f, field: :urn, label: "URN (Unique Reference Number)" do |field, options| %>
+          <%= f.text_field field, options.merge(class: 'govuk-input govuk-input--width-10') %>
+        <% end %>
+      <% end %>
```
Comment (avinhurry) [inline app/views/providers/contact.html.erb:63]: Can we put URN below the UKPRN text box please? It’s not your fault, I think the Trello ticket was slightly wrong.

Comment (stellenberger) [reply app/views/providers/contact.html.erb:63]: Of course, done! 👍

**Pull request merged**